cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc

```diff
@@ -1143,6 +1145,117 @@ struct Strftime {
 };
 #endif
 
+// ----------------------------------------------------------------------
+// Convert string representations of timestamps in arbitrary format to timestamps
+
+static std::string GetZone(std::string format) {
+  // Check for use of %z or %Z
+  size_t cur = 0;
+  std::string zone = "";
+  while (cur < format.size() - 1) {
+    if (format[cur] == '%') {
```

Comment:
Do we want to respect `%%` for escaping?

Reply:
Regex is a bit much, how about just checking if the next char is '%' and advancing the string if so? (I think we do such things elsewhere)

Reply:
Switched to a sequence counting approach.